.github/workflows/docker.yml

```
@@ -22,51 +22,76 @@ on:
       - ".pre-commit-config.yaml"
 
 jobs:
-  build-images:
-    name: Build Docker Images
+  build-test-publish-images:
+    name: Build, test, and publish Docker Images
     runs-on: ubuntu-latest
     permissions:
       contents: write
     if: >
       !contains(github.event.head_commit.message, 'ci skip') &&
       !contains(github.event.pull_request.title, 'ci skip')
+
     steps:
+      # Setup docker to build for multiple platforms, see:
+      # https://github.com/docker/build-push-action/tree/master#usage
+      # https://github.com/docker/build-push-action/blob/master/docs/advanced/multi-platform.md
+      - name: Set up QEMU (for docker buildx)
+        uses: docker/setup-qemu-action@27d0a4f181a40b142cce983c5393082c365d1480 # dependabot updates to latest release
+
+      - name: Set up Docker Buildx (for multi-arch builds)
+        uses: docker/setup-buildx-action@0d135e0c2fc0dba0729c1a47ecfcf5a3c7f8579e # dependabot updates to latest release
+        with:
+          # Allows pushing to registry on localhost:5000
+          driver-opts: network=host
```

```suggestion
        with:
          # Allows pushing to registry on localhost:5000
          driver-opts: network=host
```

Ah we ended up not using this btw.